### PR TITLE
Fix: added Resource validation for wildcard address and enabled browser shortcut

### PR DESCRIPTION
**Pull request by vmanilo**
Resolves issue with wildcard address

## Changes
- added validation on Resource create and update

### CODE REVIEW

File: twingate/internal/provider/resource/resource.go [hunk 1]
```diff
@@ -17,9 +17,10 @@ import (
 )
 
 var (
-	ErrPortsWithPolicyAllowAll      = errors.New(model.PolicyAllowAll + " policy does not allow specifying ports.")
-	ErrPortsWithPolicyDenyAll       = errors.New(model.PolicyDenyAll + " policy does not allow specifying ports.")
-	ErrPolicyRestrictedWithoutPorts = errors.New(model.PolicyRestricted + " policy requires specifying ports.")
+	ErrPortsWithPolicyAllowAll            = errors.New(model.PolicyAllowAll + " policy does not allow specifying ports.")
+	ErrPortsWithPolicyDenyAll             = errors.New(model.PolicyDenyAll + " policy does not allow specifying ports.")
+	ErrPolicyRestrictedWithoutPorts       = errors.New(model.PolicyRestricted + " policy requires specifying ports.")
+	ErrWildcardAddressWithEnabledShortcut = errors.New("not allowed to have both: wildcard address and enabled browser shortcut, please choose one of it")
```
Comment (vmanilo) [inline twingate/internal/provider/resource/resource.go:23]: @ekampf @alexb-twingate here is error placeholder message

File: twingate/internal/provider/resource/resource.go [hunk 2]
```diff
@@ -497,13 +498,24 @@ func convertResource(data *schema.ResourceData) (*model.Resource, error) {
 	}
 
 	isBrowserShortcutEnabled, ok := data.GetOkExists(attr.IsBrowserShortcutEnabled) //nolint
-	if val := isBrowserShortcutEnabled.(bool); ok {
+	if val := isBrowserShortcutEnabled.(bool); ok && isAttrKnown(data, attr.IsBrowserShortcutEnabled) {
 		res.IsBrowserShortcutEnabled = &val
 	}
 
+	if res.IsBrowserShortcutEnabled != nil && *res.IsBrowserShortcutEnabled && strings.HasPrefix(res.Address, "*.") {
```
Comment (alexb-twingate) [inline twingate/internal/provider/resource/resource.go:505]: Wildcard addresses are more than just the address being prefixed with `*.`. Wildcard addresses include:
- An address that has a wildcard, which is `*` and `?`, e.g. `redis-?-blah.internal` or `redis-*-blah.internal`
- CIDR ranges, e.g. `10.0.0.0/16`

File: twingate/internal/provider/resource/resource.go (hunk 1 above)
Comment (alexb-twingate) [reply twingate/internal/provider/resource/resource.go:23]: Here's my take: "Resource with a CIDR range or wildcard can't have the browser shortcut enabled."

Comment (vmanilo) [reply twingate/internal/provider/resource/resource.go]: thank you, update error message

File: twingate/internal/provider/resource/resource.go (hunk 2 above)
Comment (vmanilo) [reply twingate/internal/provider/resource/resource.go]: added test cases

**Review by ekampf — APPROVED** [review]

**Pull request merged**